### PR TITLE
♻️Pydantic V2:  Unify errors dv 2

### LINKED ISSUE #6756: Unexpected keyword argument when constructing errors instances (Director-v2)
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Which deploy/s?

_No response_

### Current Behavior

Pylint complains with a bunch of `E1123: Unexpected keyword argument` errors when invoking the constructor of custom errors:

```
Linting /home/runner/work/osparc-simcore/osparc-simcore/services/director-v2
Makefile:47: warning: overriding recipe for target '.env'
../../scripts/common.Makefile:81: warning: ignoring old recipe for target '.env'
# Copying makefile from .env-devel
Makefile:47: warning: overriding recipe for target '.env'
../../scripts/common.Makefile:81: warning: ignoring old recipe for target '.env'
Using config file /home/runner/work/osparc-simcore/osparc-simcore/.pylintrc
************* Module simcore_service_director_v2.modules.dask_clients_pool
src/simcore_service_director_v2/modules/dask_clients_pool.py:111:18: E1123: Unexpected keyword argument 'cluster' in constructor call (unexpected-keyword-arg)
src/simcore_service_director_v2/modules/dask_clients_pool.py:111:18: E1123: Unexpected keyword argument 'error' in constructor call (unexpected-keyword-arg)
************* Module simcore_service_director_v2.modules.dask_client
src/simcore_service_director_v2/modules/dask_client.py:525:18: E1123: Unexpected keyword argument 'job_id' in constructor call (unexpected-keyword-arg)
************* Module simcore_service_director_v2.modules.clusters_keeper
src/simcore_service_director_v2/modules/clusters_keeper.py:37:18: E1123: Unexpected keyword argument 'eta' in constructor call (unexpected-keyword-arg)
src/simcore_service_director_v2/modules/clusters_keeper.py:41:18: E1123: Unexpected keyword argument 'eta' in constructor call (unexpected-keyword-arg)
************* Module simcore_service_director_v2.modules.db.repositories.clusters
src/simcore_service_director_v2/modules/db/repositories/clusters.py:162:22: E1123: Unexpected keyword argument 'cluster_id' in constructor call (unexpected-keyword-arg)
src/simcore_service_director_v2/modules/db/repositories/clusters.py:174:18: E1123: Unexpected keyword argument 'cluster_id' in constructor call (unexpected-keyword-arg)
src/simcore_service_director_v2/modules/db/repositories/clusters.py:186:22: E1123: Unexpected keyword argument 'cluster_id' in constructor call (unexpected-keyword-arg)
src/simcore_service_director_v2/modules/db/repositories/clusters.py:199:22: E1123: Unexpected keyword argument 'cluster_id' in constructor call (unexpected-keyword-arg)
src/simcore_service_director_v2/modules/db/repositories/clusters.py:204:26: E1123: Unexpected keyword argument 'cluster_id' in constructor call (unexpected-keyword-arg)
src/simcore_service_director_v2/modules/db/repositories/clusters.py:228:34: E1123: Unexpected keyword argument 'cluster_id' in constructor call (unexpected-keyword-arg)
src/simcore_service_director_v2/modules/db/repositories/clusters.py:238:26: E1123: Unexpected keyword argument 'cluster_id' in constructor call (unexpected-keyword-arg)
src/simcore_service_director_v2/modules/db/repositories/clusters.py:248:22: E1123: Unexpected keyword argument 'cluster_id' in constructor call (unexpected-keyword-arg)
src/simcore_service_director_v2/modules/db/repositories/clusters.py:266:22: E1123: Unexpected keyword argument 'cluster_id' in constructor call (unexpected-keyword-arg)
src/simcore_service_director_v2/modules/db/repositories/clusters.py:273:22: E1123: Unexpected keyword argument 'cluster_id' in constructor call (unexpected-keyword-arg)
src/simcore_service_director_v2/modules/db/repositories/clusters.py:285:22: E1123: Unexpected keyword argument 'cluster_id' in constructor call (unexpected-keyword-arg)
************* Module simcore_service_director_v2.modules.db.repositories.comp_runs
src/simcore_service_director_v2/modules/db/repositories/comp_runs.py:120:18: E1123: Unexpected keyword argument 'cluster_id' in constructor call (unexpected-keyword-arg)
************* Module simcore_service_director_v2.utils.dask
src/simcore_service_director_v2/utils/dask.py:481:14: E1123: Unexpected keyword argument 'original_scheduler_id' in constructor call (unexpected-keyword-arg)
src/simcore_service_director_v2/utils/dask.py:481:14: E1123: Unexpected keyword argument 'current_scheduler_id' in constructor call (unexpected-keyword-arg)
src/simcore_service_director_v2/utils/dask.py:488:14: E1123: Unexpected keyword argument 'original_scheduler_id' in constructor call (unexpected-keyword-arg)
src/simcore_service_director_v2/utils/dask.py:488:14: E1123: Unexpected keyword argument 'current_scheduler_id' in constructor call (unexpected-keyword-arg)
************* Module simcore_service_director_v2.utils.dask_client_utils
src/simcore_service_director_v2/utils/dask_client_utils.py:[161](https://github.com/ITISFoundation/osparc-simcore/actions/runs/11908248587/job/33183374969#step:8:162):14: E1123: Unexpected keyword argument 'endpoint' in constructor call (unexpected-keyword-arg)
src/simcore_service_director_v2/utils/dask_client_utils.py:161:14: E1123: Unexpected keyword argument 'error' in constructor call (unexpected-keyword-arg)
src/simcore_service_director_v2/utils/dask_client_utils.py:[164](https://github.com/ITISFoundation/osparc-simcore/actions/runs/11908248587/job/33183374969#step:8:165):14: E1123: Unexpected keyword argument 'endpoint' in constructor call (unexpected-keyword-arg)
src/simcore_service_director_v2/utils/dask_client_utils.py:164:14: E1123: Unexpected keyword argument 'error' in constructor call (unexpected-keyword-arg)
src/simcore_service_director_v2/utils/dask_client_utils.py:167:14: E1123: Unexpected keyword argument 'endpoint' in constructor call (unexpected-keyword-arg)
src/simcore_service_director_v2/utils/dask_client_utils.py:[167](https://github.com/ITISFoundation/osparc-simcore/actions/runs/11908248587/job/33183374969#step:8:168):14: E1123: Unexpected keyword argument 'error' in constructor call (unexpected-keyword-arg)
************* Module test_modules_comp_scheduler_dask_scheduler
tests/unit/with_dbs/test_modules_comp_scheduler_dask_scheduler.py:1097:8: E1123: Unexpected keyword argument 'original_scheduler_id' in constructor call (unexpected-keyword-arg)
tests/unit/with_dbs/test_modules_comp_scheduler_dask_scheduler.py:1097:8: E1123: Unexpected keyword argument 'current_scheduler_id' in constructor call (unexpected-keyword-arg)
tests/unit/with_dbs/test_modules_comp_scheduler_dask_scheduler.py:1184:28: E1123: Unexpected keyword argument 'job_id' in constructor call (unexpected-keyword-arg)
tests/unit/with_dbs/test_modules_comp_scheduler_dask_scheduler.py:1220:28: E1123: Unexpected keyword argument 'job_id' in constructor call (unexpected-keyword-arg)
tests/unit/with_dbs/test_modules_comp_scheduler_dask_scheduler.py:1521:8: E1123: Unexpected keyword argument 'eta' in constructor call (unexpected-keyword-arg)
```

The issues seems in the errors hierarchy and conflicts with `__init__` method.

- Partially resolves: #6747

### Expected Behavior

No errors.

### Steps To Reproduce

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

This PR converts `DirectorError` class to inherit from `OsparcErrorMixin` and `RuntimeError`.
This leads to quite a few changes.

## Related issue/s

- Resolves #6756 

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by giancarloromeo — APPROVED** [review]

**Review by GitHK — APPROVED** [review]

**Review by GitHK — APPROVED** [review]
Looks OK to me. Strange that no tests for dynamic-sidecar broke. Please have a look if the integration tests still work as expected

File: services/director-v2/src/simcore_service_director_v2/modules/comp_scheduler/_base_scheduler.py [hunk 1]
```diff
@@ -929,22 +929,25 @@ async def _schedule_tasks_to_start(  # noqa: C901
...
             )
             await CompTasksRepository.instance(
                 self.db_engine
             ).update_project_tasks_state(
                 project_id,
-                [exc.node_id],
+                [err_context["node_id"]],
                 RunningState.FAILED,
-                exc.get_errors(),
+                None,  # exc.get_errors(), # @pcrespov I need your help here!
```
Comment (pcrespov) [inline services/director-v2/src/simcore_service_director_v2/modules/comp_scheduler/_base_scheduler.py:944]: Let's talk about this offline. In short, current `OsparcErrorMixin` does not contain this member. but there is a way around

File: services/director-v2/src/simcore_service_director_v2/api/routes/computations.py [hunk 2]
```diff
@@ -204,7 +204,9 @@ async def _get_project_node_names(
     except DBProjectNotFoundError:
         _logger.exception("Could not find project: %s", f"{project_id=}")
     except ProjectNotFoundError as exc:
-        _logger.exception("Could not find parent project: %s", f"{exc.project_id=}")
+        _logger.exception(
+            "Could not find parent project: %s", exc.error_context()["project_id"]
```
Comment (pcrespov) [inline services/director-v2/src/simcore_service_director_v2/api/routes/computations.py:208]: TIP: I would highly recommend using `create_troubleshotting_log_kwargs`
You can find a usage example in https://github.com/itisfoundation/osparc-simcore/blob/cd6f5455ab1d0e9ae874842926dab86100df07c4/services/api-server/src/simcore_service_api_server/exceptions/handlers/_handlers_factory.py#L34-L49

File: services/director-v2/src/simcore_service_director_v2/modules/comp_scheduler/_scheduler_factory.py [hunk 3]
```diff
@@ -18,7 +18,7 @@
 async def create_from_db(app: FastAPI) -> BaseCompScheduler:
     if not hasattr(app.state, "engine"):
         msg = "Database connection is missing. Please check application configuration."
-        raise ConfigurationError(msg)
+        raise ConfigurationError(msg=msg)
```
Comment (pcrespov) [inline services/director-v2/src/simcore_service_director_v2/modules/comp_scheduler/_scheduler_factory.py:21]: TIP: add context to your exceptions ...

File: services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_pipelines.py [hunk 4]
```diff
@@ -25,7 +25,7 @@ async def get_pipeline(self, project_id: ProjectID) -> CompPipelineAtDB:
             )
             row: RowProxy | None = await result.fetchone()
         if not row:
-            raise PipelineNotFoundError(str(project_id))
+            raise PipelineNotFoundError(pipeline_id=project_id)
```
Comment (pcrespov) [inline services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_pipelines.py:28]: TIP: useful context is `user_id` or `project_id`

**Review by pcrespov — APPROVED** [review]
thx. Let's talk offline about `get_error` and left some suggestions about how to raise and handle custom exceptions.

File: services/director-v2/src/simcore_service_director_v2/api/routes/computations.py (hunk 2 above)
Comment (sanderegg) [reply services/director-v2/src/simcore_service_director_v2/api/routes/computations.py:208]: only if this does not create a huge mess when I upgrade from master afterwards

File: services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_pipelines.py (hunk 4 above)
Comment (sanderegg) [reply services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_pipelines.py:28]: not sure what you mean here. this is a pydantic v2 branch.

File: services/director-v2/src/simcore_service_director_v2/modules/comp_scheduler/_base_scheduler.py (hunk 1 above)
Comment (sanderegg) [reply services/director-v2/src/simcore_service_director_v2/modules/comp_scheduler/_base_scheduler.py]: fixed after discussion

File: services/director-v2/src/simcore_service_director_v2/modules/comp_scheduler/_scheduler_factory.py (hunk 3 above)
Comment (sanderegg) [reply services/director-v2/src/simcore_service_director_v2/modules/comp_scheduler/_scheduler_factory.py:21]: same

**Review by matusdrobuliak66 — APPROVED** [review]

**Pull request merged**